### PR TITLE
Search branch

### PR DESCRIPTION
**Pull request by pablo-honeybear**
added search function and did some front end work on the top-bar

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,7 +2,11 @@ class PostsController < ApplicationController
   before_action :authenticate_user!, except: [:index, :show]
 
   def index
-    @posts = Post.all.order(created_at: :desc).page(params[:page]).per(5)
+    if params[:search]
```
Comment (pablo-honeybear) [inline app/controllers/posts_controller.rb]: We assigned @post as we intended

File: spec/features/search/user_can_search_posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+feature "user can search webseries by the webseries name" do
+  let!(:post1) { FactoryGirl.create(:post_with_three_reviews) }
+  let!(:post2) { FactoryGirl.create(:post_with_three_reviews) }
+  let!(:post3) { FactoryGirl.create(:post_with_three_reviews, webseries_name: "Pop-pop goes to prison") }
+
+  scenario "user sees a list of webseries that match their search query" do
+    visit '/'
+    # save_and_open_page
```
Comment (dreamingfurther) [inline spec/features/search/user_can_search_posts_spec.rb]: delete, don't comment these sorts of things. this is just messy

**Comment by dreamingfurther** [comment]
@pablo-honeybear fix the final comment I've made and then this looks good to merge!

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -2,7 +2,11 @@ class PostsController < ApplicationController
   before_action :authenticate_user!, except: [:index, :show]
 
   def index
```
Comment (dreamingfurther) [inline app/controllers/posts_controller.rb]: good clean controller!

Comment (pablo-honeybear) [inline app/controllers/posts_controller.rb]: thanks!

**Pull request merged**